src/pe2loaddata/transformer.py

```diff
@@ -0,0 +1,163 @@
+        dest="index_directory",
+        default=os.path.curdir,
+        help="The directory containing the index file and images"
+    )
+
+    parser.add_argument(
+        "config_file",
+        type=check_file_arg,
+        help="The config.yaml file that chooses channels and metadata for the CSV"
+    )
+
+    parser.add_argument(
+        "output_csv",
+        help="The name of the LoadData .csv file to be created"
+    )
+
+    return parser.parse_args()
+
+
+def load_config(config_file: Union[bytes, str, PathLike]) -> (Any, Any):
+    """Load the configuration from config.yaml"""
+    with open(config_file, "r") as fd:
+        config = yaml.load(fd, Loader=yaml.BaseLoader)
+
+    if isinstance(config, list):
+        config = config[0]
+
+    channels = config['channels']
+
+    metadata = config.get('metadata', {})
+
+    return channels, metadata
+
+
+def write_csv(writer, images, plates, wells, channels, metadata, paths):
+    header = sum([["_".join((prefix, channels[channel])) for prefix in ["FileName", "PathName"]] for channel in sorted(channels.keys())], [])
+
+    header += ["Metadata_Plate", "Metadata_Well", "Metadata_Site"]
+
+    header += ["_".join(("Metadata", metadata[key])) for key in sorted(metadata.keys())]
+
+    writer.writerow(header)
+
+    for plate_name in sorted(plates):
+        plate = plates[plate_name]
+
+        for well_id in plate.well_ids:
+            well = wells[well_id]
+
+            fields = {}
+
+            well_name = well.well_name
+
+            for image_id in well.image_ids:
+                try:
+                    image = images[image_id]
+
+                    # For simplifying the code, field_id is defined as the combination of
+                    # FieldID and PlaneID. Later, PlaneID is stripped out when actually
+                    # writing out field_id.
+                    field_id = '%02d-%02d' % (int(image.metadata["FieldID"]), int(image.metadata.get("PlaneID", 1)))
+
+                    channel = image.channel_name
+
+                    assert channel in channels
+
+                    if field_id not in fields:
+                        fields[field_id] = {channel: image}
+                    else:
+                        fields[field_id][channel] = image
+                except Exception as e:
+                    print(e)
+
+            for field in sorted(fields):
+                d = fields[field]
+                row = []
+
+                for channel in sorted(channels.keys()):
+                    try:
+                        image = d[channel]
+
+                        file_name = image.metadata["URL"]
+
+                        row += [file_name, paths[file_name]]
+                    except Exception as e:
+                        logging.debug("Channel = {}; Field = {}; Well = {}; Well_id = {}; Plate = {}".format(channel, field, plate_name))
+
+                        print(e)
+
+                        row = []
+
+                        break
```

Comment:
Why are we breaking here?  Wouldn't it be better to just continue?

Reply:
It should break on `KeyError` right? i.e. when the channel is not present. 

The user should do this 
https://github.com/cytomining/profiling-handbook/blob/b9647f8845cdcfbe2c0c7ec5aa3c4f8695e40433/03-setup-pipelines.Rmd#L183-L188

but it would certainly be nice if pe2loaddata was more helpful here.

Reply:
It means that even a single missing file (which happens a non-zero fraction of the time) breaks the whole creation script, as opposed to just skipping that line.  I suppose we could make it a configurable option to pass vs break...